src/control/server/drpc.go

```diff
@@ -46,17 +46,26 @@ func getDrpcClientConnection(sockDir string) *drpc.ClientConnection {
 // drpcSetup creates socket directory, specifies socket path and then
 // starts drpc server.
 func drpcSetup(sockDir string, iosrv *iosrv) error {
```

Comment:
It would be simpler to break out the management of the sockDir into its own function which is solely concerned with the sockDir. Following the pattern of returning as early as possible also tends to shorten and simplify code paths.

```go
func createSocketDir(sockDir string) error {
	// FIXME: shouldn't there be some checks on the socket dir even if it exists?
	// e.g. permissions, that it's actually a directory and not a file, etc?
	_, err := os.Stat(sockDir)
	if err == nil {
		return nil
	}

	switch {
	case os.IsPermission(err):
		return errors.Wrapf(err, "user does not have permission to access %s", sockDir)
	case os.IsNotExist(err):
		if err := os.MkdirAll(sockDir, 0755); err != nil {
			return errors.Wrapf(err, "unable to create socket directory %s", sockDir)
		}
	}

	return errors.Wrapf(err, "unknown error while creating sockDir %s", sockDir)
}

// drpcSetup creates socket directory, specifies socket path and then
// starts drpc server.
func drpcSetup(sockDir string, iosrv *iosrv) error {
	if err := createSocketDir(sockDir); err != nil {
		// probably no value added by wrapping here
		return err
	}
...
```

Reply:
sounds good but maybe should be in its own patch? if you agree could you create a ticket and just paste this comment please

Reply:
nvm, made the changes